src/components/todo-item/TodoItem.tsx

```diff
@@ -0,0 +1,98 @@
+"use client";
+
+import React, { useState } from "react";
+import styles from "./Index.module.css";
+import { Checkbox } from "../checkbox/Checkbox";
+import classnames from "classnames";
+import { HiMiniTrash } from "react-icons/hi2";
+
+interface TodoItemProps {
+  item: ITodoItem;
```

Comment:
As discussed before, I think it's better to split up the props like this:

```
  title: string;
  completed: boolean;
```

This makes the component more reusable on not depending on a full `ITodoItem`